### PR TITLE
Vcf gz datatype

### PR DESCRIPTION
**Pull request by ffinfo**

### CODE REVIEW

**Comment by mvdbeek** [comment]
@galaxybot test this

File: lib/galaxy/datatypes/tabular.py [hunk 1]
```diff
@@ -62,15 +63,15 @@ def displayable( self, dataset ):
             return False
 
     def get_chunk(self, trans, dataset, offset=0, ck_size=None):
-        with open(dataset.file_name) as f:
-            f.seek(offset)
-            ck_data = f.read(ck_size or trans.app.config.display_chunk_size)
-            if ck_data and ck_data[-1] != '\n':
+        f = compression_utils.get_fileobj(dataset.file_name)
```
Comment (mvdbeek) [inline lib/galaxy/datatypes/tabular.py]: I think we can do `with compression_utils.get_fileobj(dataset.file_name) as f:`.

**Comment by mvdbeek** [comment]
@ffinfo let me know if you want to continue working on this (I think we're almost there), otherwise I can do the last bit of polishing.

**Comment by ffinfo** [comment]
@mvdbeek Ye was still planning to continue on this. I think I still have some uncommitted code for a test tool. Was a bit of a chaos when I did came back to Leiden. You know how that goes with todo's ;)

File: test/functional/tools/vcf_bgzip.xml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      md5sum input.vcf.gz.tbi > input.vcf.gz.tbi.md5 &&
+      cp $input.metadata.vcf_gz_index input.vcf.gz.tbi &&
+      md5check -c vcf.gz.tbi.md5 &&
+      cp $input $output
+    ]]></command>
+    <inputs>
+        <param name="input" format="vcf_bgzip" type="data" label="Source file"/>
+    </inputs>
+    <outputs>
+        <data format="data" ftype="vcf_bgzip" name="output" />
```
Comment (mvdbeek) [inline test/functional/tools/vcf_bgzip.xml]: This should be `format="vcf_bgzip"`. ftype is (confusingly ...) only used in the test section.

**Comment by mvdbeek** [comment]
@galaxybot test this

**Comment by mvdbeek** [comment]
@galaxybot test this

**Comment by mvdbeek** [comment]
@ffinfo this is looking good to me, If you're happy with this I can merge it. bctools generates csi indexes by default (i.e without the `-t` flag), are these widely used ? Should we also implement them ?

**Comment by ffinfo** [comment]
Well tools like Picard and GATK are using by default '.tbi' extensions. The csi is more used for bcf files and not for vcf files. That why I think it's better to keep it with the '-t' flag

**Comment by mvdbeek** [comment]
Alright, we can always add this if there is demand.

**Comment by mvdbeek** [comment]
@galaxybot test this

**Comment by mvdbeek** [comment]
@galaxybot test this

**Comment by jmchilton** [comment]
Awesome - thanks for the contribution @ffinfo and @mvdbeek !

**Pull request merged**